@here/harp-materials/lib/SolidLineMaterial.ts

```diff
@@ -308,16 +401,41 @@ export class SolidLineMaterial extends THREE.RawShaderMaterial
         this.uniforms.diffuse.value = value;
     }
 
+    get dashColor(): THREE.Color {
+        return this.uniforms.dashColor.value as THREE.Color;
+    }
+    set dashColor(value: THREE.Color) {
+        this.uniforms.dashColor.value = value;
```

Comment:
You can do
```suggestion
        this.uniforms.dashColor.value.copy(value);
```
here, and then just assign `dashColor = color` without weird `this.dashColor = this.dashColor;` stuff

Reply:
@okanishchev I guess you mean `this.dashColor.set(params.dashColor)` instead of `this.dashColor = this.dashColor;`. as copy() doesn't remove the need of calling setter

Reply:
@okanishchev That makes it even worse as if you assign new color and then change it outside, it wouldn't work!!!. + It doesn't help solve the setter issue in any way.

Example:
```js
const myColor = new THREE.Color();
solidLineMaterial.dashColor = myColor;
myColor.set(0xff0000); // wouldn't work as dashColor was copied
```

Reply:
Then you should call the `updateDashAlpha()` function explicitly when it's about to change, or add special property setter for this and also assign it explicitly.
Or you can even use `onBeforeRender` callback to update this values, but I'm not sure if it will not affect performance.

Reply:
@okanishchev How is it related to the written above?

Reply:
Still not perfect solution, and I think you should leave some comments why it's done this way - enabling dash color only if you've passed a style for that.